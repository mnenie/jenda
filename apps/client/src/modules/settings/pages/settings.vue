<script setup lang="ts">
import { useHead } from '@unhead/vue'

definePage({
  meta: {
    requiresAuth: true,
  },
  name: 'settings',
})

useHead({
  title: 'Jenda | Settings',
})
</script>

<template>
  <div>
    settings
  </div>
</template>
