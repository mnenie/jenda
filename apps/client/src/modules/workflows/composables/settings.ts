import { inject, type InjectionKey, provide } from 'vue'
import type { SelectedNodeContext } from '../types/contexts'

const key: InjectionKey<SelectedNodeContext> = Symbol('selected-node')

export function provideSelectedNodeContext<T extends SelectedNodeContext>(value: T) {
  provide(key, value)
}

export function useSelectedNodeContext() {
  return inject(key)!
}
