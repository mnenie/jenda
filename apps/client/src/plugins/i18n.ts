import { createI18n } from 'vue-i18n'
import { enZod, ruZod, zhZod } from '@/shared/libs/vee-validate/rules'
import { customPluralRule } from '@/shared/libs/i18n/plurals'
import { getCurrentLocale } from '@/shared/libs/i18n/utils/getLocale'

interface MessageSchema {
  [key: string]: any
}

const ru_ru = import.meta.glob(
  '@/shared/libs/i18n/locales/**/ru-RU.ts',
  { eager: true },
)
const en_us = import.meta.glob(
  '@/shared/libs/i18n/locales/**/en-US.ts',
  { eager: true },
)
const zh_cn = import.meta.glob(
  '@/shared/libs/i18n/locales/**/zh-CN.ts',
  { eager: true },
)

function loadLang(modules: Record<string, any>) {
  const messages: { [key: string]: string } = {}

  Object.keys(modules).forEach((module) => {
    Object.assign(messages, { ...modules[module].default })
  })
  return messages
}

const messages = {
  'en-US': {
    ...loadLang(en_us),
    errors: {
      ...enZod,
    },
  },
  'ru-RU': {
    ...loadLang(ru_ru),
    errors: {
      ...ruZod,
    },
  },
  'zh-CN': {
    ...loadLang(zh_cn),
    errors: {
      ...zhZod,
    },
  },
}

const i18n = createI18n<[MessageSchema], 'en-US' | 'ru-RU' | 'zh-CN'>({
  legacy: false,
  locale: getCurrentLocale.value,
  globalInjection: true,
  pluralRules: {
    'ru-RU': {
      // @ts-expect-error i18n shema
      customPluralRule,
    },
  },
  messages,
})

export default i18n
