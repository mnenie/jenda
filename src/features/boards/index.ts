export * from './add-board/ui';
export * from './charts/ui';
export * from './filter/ui';
