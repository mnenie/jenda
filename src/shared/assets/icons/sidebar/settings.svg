<svg xmlns="http://www.w3.org/2000/svg" width="1em" height="1em" viewBox="0 0 24 24"><g fill="none" stroke="currentColor" stroke-linecap="round" stroke-linejoin="round" stroke-width="1.5" color="currentColor"><path d="m21.318 7.141l-.494-.856c-.373-.648-.56-.972-.878-1.101c-.317-.13-.676-.027-1.395.176l-1.22.344c-.459.106-.94.046-1.358-.17l-.337-.194a2 2 0 0 1-.788-.967l-.334-.998c-.22-.66-.33-.99-.591-1.178c-.261-.19-.609-.19-1.303-.19h-1.115c-.694 0-1.041 0-1.303.19c-.261.188-.37.518-.59 1.178l-.334.998a2 2 0 0 1-.789.967l-.337.195c-.418.215-.9.275-1.358.17l-1.22-.345c-.719-.203-1.078-.305-1.395-.176c-.318.129-.505.453-.878 1.1l-.493.857c-.35.608-.525.911-.491 1.234c.034.324.268.584.736 1.105l1.031 1.153c.252.319.431.875.431 1.375s-.179 1.056-.43 1.375l-1.032 1.152c-.468.521-.702.782-.736 1.105s.14.627.49 1.234l.494.857c.373.647.56.971.878 1.1s.676.028 1.395-.176l1.22-.344a2 2 0 0 1 1.359.17l.336.194c.36.23.636.57.788.968l.334.997c.22.66.33.99.591 1.18c.262.188.609.188 1.303.188h1.115c.694 0 1.042 0 1.303-.189s.371-.519.59-1.179l.335-.997c.152-.399.428-.738.788-.968l.336-.194c.42-.215.9-.276 1.36-.17l1.22.344c.718.204 1.077.306 1.394.177c.318-.13.505-.454.878-1.101l.493-.857c.35-.607.525-.91.491-1.234s-.268-.584-.736-1.105l-1.031-1.152c-.252-.32-.431-.875-.431-1.375s.179-1.056.43-1.375l1.032-1.153c.468-.52.702-.781.736-1.105s-.14-.626-.49-1.234"/><path d="M15.52 12a3.5 3.5 0 1 1-7 0a3.5 3.5 0 0 1 7 0"/></g></svg>