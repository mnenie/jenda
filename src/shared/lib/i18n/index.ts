import { createI18n } from 'vue-i18n';
import { useCookies } from '@vueuse/integrations/useCookies';
import enLocale from './locales/en-US';
import ruLocale from './locales/ru-RU';
import zhLocale from './locales/zh-CN';

type MessageSchema = typeof enLocale;

const messages = {
  'en-US': {
    ...enLocale
  },
  'ru-RU': {
    ...ruLocale
  },
  'zh-CN': {
    ...zhLocale
  }
};
const cookies = useCookies();

const getCurrentLocale = () => {
  const cookieLanguage = cookies.get('i18n');
  if (cookieLanguage) {
    return cookieLanguage;
  }

  return 'en-US';
};

const i18n = createI18n<[MessageSchema], 'en-US' | 'ru-RU' | 'zh-CN'>({
  legacy: false,
  locale: getCurrentLocale(),
  messages: messages
});

export default i18n;
