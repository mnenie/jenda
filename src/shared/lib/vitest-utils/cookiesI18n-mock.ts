import { vi } from 'vitest';

vi.mock('@vueuse/integrations/useCookies', () => {
  return {
    useCookies: () => ({
      get(key: string) {
        return key === 'i18n' ? 'en-US' : undefined;
      }
    })
  };
});
