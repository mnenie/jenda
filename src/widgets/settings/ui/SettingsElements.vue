<script setup lang="ts">
import { ThemeSwitcher } from '@/features/settings/theme-switcher';
import { LanguageSwitcher } from '@/features/settings/lang-switcher';
</script>

<template>
  <div :class="$style.lang">
    <p class="text-sm" style="font-weight: 500">{{ $t('settings.lang.label') }}</p>
    <LanguageSwitcher />
    <span class="text-xs"> {{ $t('settings.lang.about') }} </span>
  </div>
  <div :class="$style.theme">
    <div :class="$style.text">
      <p class="text-sm" style="font-weight: 500">{{ $t('settings.theme.label') }}</p>
      <span class="text-xs">{{ $t('settings.theme.about') }}</span>
    </div>
    <ThemeSwitcher />
  </div>
</template>

<style module lang="scss">
.lang {
  display: flex;
  flex-direction: column;
  gap: 6px;
  margin-bottom: 40px;
  align-items: flex-start;

  & > p {
    color: var(--zinc-950);
  }

  & > span {
    color: var(--zinc-500);
  }
}

.theme {
  .text {
    margin-bottom: 12px;

    & p {
      color: var(--zinc-950);
    }

    & span {
      color: var(--zinc-500);
    }
  }
}

:global(html.dark) {
  .lang {
    & > p {
      color: var(--zinc-200);
    }
    & > span {
      color: var(--zinc-300);
    }
  }
  .theme {
    .text {
      & p {
        color: var(--zinc-200);
      }
      & span {
        color: var(--zinc-300);
      }
    }
  }
}
</style>
