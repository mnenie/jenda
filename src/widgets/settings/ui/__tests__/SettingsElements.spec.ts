import { describe, it, expect } from 'vitest';
import { shallowMount } from '@vue/test-utils';
import '@/shared/lib/vitest-utils/cookiesI18n-mock';
import SettingsElements from '../SettingsElements.vue';
import i18n from '@/shared/lib/i18n';
import { LanguageSwitcher, ThemeSwitcher } from '@/features/settings';

describe('tests for SettingsElements.vue', () => {
  const wrapper = shallowMount(SettingsElements, {
    global: {
      plugins: [i18n]
    }
  });

  it('should be render correctly', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should render subcomponents', () => {
    expect(wrapper.findComponent(LanguageSwitcher).exists()).toBe(true);
    expect(wrapper.findComponent(ThemeSwitcher).exists()).toBe(true);
  });

  it('should has correctly localization', () => {
    const textLang = wrapper.find('.lang p');
    expect(textLang.text()).toContain('Language');

    const textTheme = wrapper.find('.theme p');
    expect(textTheme.text()).toContain('Theme');
  });
});
