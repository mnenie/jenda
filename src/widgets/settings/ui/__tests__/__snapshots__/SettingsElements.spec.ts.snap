// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`tests for SettingsElements.vue > should be render correctly 1`] = `
"<div class="lang">
  <p class="text-sm" style="font-weight: 500;">Language</p>
  <language-switcher-stub></language-switcher-stub><span class="text-xs">This is the language that will be used in the dashboard.</span>
</div>
<div class="theme">
  <div class="text">
    <p class="text-sm" style="font-weight: 500;">Theme</p><span class="text-xs">Select the theme for the dashboard.</span>
  </div>
  <theme-switcher-stub></theme-switcher-stub>
</div>"
`;
